File: testData/src/java8/pkg/TestTryLoop.java
package pkg;

import java.io.File;
import java.io.FileNotFoundException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Iterator;
import java.util.Scanner;
import java.util.ServiceConfigurationError;

public class TestTryLoop {
  private boolean field;

  public void test(File file) {
    try {
      while (this.field) {
        Scanner scanner = new Scanner(file);
      }
    } catch (FileNotFoundException e) {
      e.printStackTrace();
    }
  }

  public boolean hasNext(Path p, Iterator<File> f) {
    File a;
    while (true) {
      try {
        if (Files.exists(p)) {
          a = f.next();
          return true;
        } else
          return false;
      } catch (ServiceConfigurationError e) {
        System.out.println(1);
      } catch (NoClassDefFoundError e) {
        System.out.println(2);
      }
    }
  }
}

File: plugins/kotlin/testData/src/kt/pkg/TestTryLoop.kt
package pkg

import java.io.File
import java.io.FileNotFoundException
import java.nio.file.Files
import java.nio.file.Path
import java.util.*

class TestTryLoop {
    private val field = false

    fun test(file: File) {
        try {
            while (this.field) {
                val scanner = Scanner(file)
            }
        } catch (e: FileNotFoundException) {
            e.printStackTrace()
        }
    }

    fun hasNext(p: Path, f: MutableIterator<File?>): Boolean {
        var a: File?
        while (true) {
            try {
                if (Files.exists(p)) {
                    a = f.next()
                    return true
                } else return false
            } catch (e: ServiceConfigurationError) {
                println(1)
            } catch (e: NoClassDefFoundError) {
                println(2)
            }
        }
    }
}
